core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataStatus.java

```diff
@@ -55,27 +56,42 @@ public boolean isUserMetadataStatus(int userId) throws Exception {
     }
 
     /**
-     * Return all status records for the metadata id - current status is the first child due to sort by DESC on changeDate
+     * Return last workflow status for the metadata id
      */
     @Override
     public MetadataStatus getStatus(int metadataId) throws Exception {
         String sortField = SortUtils.createPath(MetadataStatus_.id, MetadataStatusId_.changeDate);
-        List<MetadataStatus> status = metadataStatusRepository.findAllById_MetadataId(metadataId, new Sort(Sort.Direction.DESC, sortField));
+        List<MetadataStatus> status = metadataStatusRepository.findAllByIdAndByType(
```

Comment:
Related to the PR to add database field indexes #3359, check to add in `MetadataStatus` a composed index for metadataid and type.

Reply:
> Related to the PR to add database field indexes #3359, check to add in `MetadataStatus` a composed index for metadataid and type.

From `org/hibernate/javax/persistence/hibernate-jpa-2.1-api/1.0.0.Final/hibernate-jpa-2.1-api-1.0.0.Final-sources.jar!/javax/persistence/Index.java`
```
Note that it is not necessary to specify an index
 * for a primary key, as the primary key index will be created automatically
```

and we have currently a correct PK

```
  CONSTRAINT metadatastatus_pkey PRIMARY KEY (changedate, metadataid, statusid, userid),
```

So that would be useless in that case?

Maybe an index on the ```type``` column in statusvalues ?

Reply:
Afaik the PK contains an index for all these fields: changedate, metadataid, statusid, userid, but will not be used if only searching by metadataid

Reply:
Added in https://github.com/geonetwork/core-geonetwork/commit/275832295681349d3d84e709e04874f36237f613

For the record, 
* adding `@Index` annotations will be take into account on next startup of the app and indexes created
* pgadmin filter out index on PK or FK (use \d in psql for the details)